src/blocks/opentable/edit.js

```diff
@@ -0,0 +1,190 @@
+				'&query=' +
+				encodeURIComponent( token )
+		)
+			.then( ( response ) => response.json() )
+			.catch( ( error ) => {
+				setIsLoading( false );
+				noticeOperations.createErrorNotice( __( 'Error connecting to the OpenTable API. Please try again later.', 'coblocks' ) );
+				throw new Error( 'Error connecting to the OpenTable API: ' + error );
+			} )
+			.then( ( json ) => {
+				const results = [];
+				for ( const item in json.items ) {
+					const itemProps = json.items[ item ];
+					const name = `${ decodeURIComponent( itemProps.name ) },  ${ decodeURIComponent( itemProps.addressResponse.city ) }, ${ decodeURIComponent( itemProps.addressResponse.country ) } (${ itemProps.rid })`;
+					results.push( name );
+				}
+				setQueryResults( results );
+				setNoResultsFound( results?.length === 0 );
+				setIsLoading( false );
+			} );
+	}, 500 );
+
+	useEffect( () => {
+		if ( attributes.language === '' ) {
+			switch ( getLocaleCode().substring( 0, coblocksBlockData.localeCode.indexOf( '_' ) ) ) {
+				case 'fr':
+					props.setAttributes( { language: 'fr-CA' } );
+					break;
+				case 'de':
+					props.setAttributes( { language: 'de-DE' } );
+					break;
+				case 'es':
+					props.setAttributes( { language: 'es-MX' } );
+					break;
+				case 'ja':
+					props.setAttributes( { language: 'ja-JP' } );
+					break;
+				case 'nl':
+					props.setAttributes( { language: 'nl-NL' } );
+					break;
+				case 'it':
+					props.setAttributes( { language: 'it-IT' } );
+					break;
+				default:
+					props.setAttributes( { language: 'en-US' } );
+					break;
+			}
+		}
+	}, [] );
+
+	return (
+		<>
+			<Controls
+				isEditing={ isEditing }
+				setIsEditing={ setIsEditing }
+				{ ...props } />
+			<InspectorControls
+				className={ className }
+				attributes={ attributes }
+				setAttributes={ props.setAttributes }
+			/>
+			<div className={ className }>
+
+				{ ( isEditing ) ? (
+					<Placeholder
+						icon={ <Icon icon={ icon } /> }
+						label={ __( 'OpenTable', 'coblocks' ) }
+						instructions={ __(
+							'Select your OpenTable restaurant(s) to show the reservations widget.',
+							'coblocks'
+						) }
+						isColumnLayout={ true }
+						notices={
+							noticeUI
+						}
+					>
+
+						<div className="components-placeholder__flex-fields">
+							<div className="components-placeholder__flex-fields-vertical-group">
+								<FormTokenField
+									value={ ridField || '' }
+									suggestions={ queryResults }
+									className="components-placeholder__input"
+									placeholder={ __(
+										'Select restaurant(s)',
+										'coblocks'
+									) }
+									maxSuggestions={ 15 }
+									onChange={ setRidField }
+									onInputChange={ searchRestaurants }
+									__experimentalHowTo={ false }
+									tokenizeOnSpace={ false }
+								/>
+								<Button
+									isPrimary={ !! ridField }
+									isSecondary={ ! ridField }
+									type="submit"
+									disabled={ ridField.length < 1 }
```

Comment:
This logic does not work as expected due to the way in which JavaScript evaluates empty arrays.
Here are some evaluation examples.:
```javascript
isPrimary={ !! ridField }
 > const ridField = [] 
 > !! ridField | true
 
  > const ridField = [ 123, 212 ] 
  > !! ridField | true
  
 isSecondary={ ! ridField }
  > const ridField = [] 
 > ! ridField | false
 
  > const ridField = [ 123, 212 ] 
  > ! ridField | false
```

This is because an empty array is seen as truthy value in JavaScript. In this case, it would be wise to use the `!!` operator with the array.length property. 

```javascript
!! 0 | false
!! 1 | true

! 0 | true
! 1 | false
```

As a result, it looks like the button only has the `isPrimary` prop and not the `isSecondary`. Here is a screenshot of the secondary style applied and it looks like it is missing some margin/padding.
![image](https://user-images.githubusercontent.com/30462574/129060831-7c568bbb-fc93-4971-8efd-0e08dae96f43.png)

Reply:
I went ahead and completely removed this logic so that the button has a constant `isPrimary` prop instead. The primary styles have all been handled. If you would like to introduce styles for the `isSecondary` button state and add the logic back I am fine with that too.